### PR TITLE
Add highlighting for yield keyword from #[feature(generators)]

### PR DESCRIPTION
**Pull request by Keating950**
This PR adds highlighting for `yield` as a keyword to support the [generator feature](https://doc.rust-lang.org/beta/unstable-book/language-features/generators.html) found in nightly Rust.

NOTE: This currently causes `tree-sitter test` to fail with the message ` Query error at 102:2. Invalid node type yield`. I'm a bit  confused as to why, given that it's declared in the same way as all the other keywords around it; do you have any suggestions as to how to fix it?

### CODE REVIEW

**Comment by maxbrunsfeld** [comment]
For this to work, `yield` needs to be a token in the grammar. I'm open to adding it if you want to update the PR to add parsing for `yield` statements.

**Comment by Keating950** [comment]
After doing some reading on how tree-sitter parsers are written, I can definitely see why my first commit didn't work... oops. I've got it working properly now. Would you prefer that I squash my working local copy with the commit in this PR or just push it and leave both commits?

**Comment by Keating950** [comment]
Hey @maxbrunsfeld, any suggestions re: the questions above?

**Comment by maxbrunsfeld** [comment]
You can do either. I usually merge PRs with a squash, so it won't matter how many commits are on the branch.

**Comment by Keating950** [comment]
All done @maxbrunsfeld.

**Comment by aryx** [comment]
would you mind rebasing @Keating950 ?

**Comment by Keating950** [comment]
No problem, all done.

File: grammar.js [hunk 1]
```diff
@@ -988,6 +989,11 @@ module.exports = grammar({
       prec(-1, 'return'),
     ),
 
+    yield_expression: $ => choice(
+      prec.left(seq('yield', $._expression)),
+      prec(-1, 'yield'),
```
Comment (aryx) [inline grammar.js:994]: would be nice to comment why we need a prec(-1) here, which ambiguity does it resolve?
with example of code showing the problem.

**Review by aryx — APPROVED** [review]

**Pull request merged**